content_resolver/analyzer.py

```diff
@@ -15,6 +17,315 @@ def pkg_placeholder_name_to_nevr(placeholder_name):
+                #
+                # So the usual line has file_line.split() == 8
+                # The one with the long package name has file_line.split() == 3
+                # and the one following it has file_line.split() == 7
+                # 
+                # One more thing... long release!
+                #
+                # DEBUG util.py:446:   qrencode-devel               aarch64 4.0.2-8.fc35                  build  13 k
+                # DEBUG util.py:446:   systemtap-sdt-devel          aarch64 4.6~pre16291338gf2c14776-1.fc36
+                # DEBUG util.py:446:                                                                      build  71 k
+                # DEBUG util.py:446:   tpm2-tss-devel               aarch64 3.1.0-4.fc36                  build 315 k
+                #
+                # So the good one here is file_line.split() == 5.
+                # And the following is also file_line.split() == 5. Fun!
+                # 
+                # So if it ends with B, k, M, G it's the wrong line, so skip, otherwise take the package name.
+                #
+                # I can also anticipate both get long... that would mean I need to skip file_line.split() == 4.
+
+                if len(file_line.split()) == 10 or len(file_line.split()) == 11:
+                    # Sometimes DNF5 prints "Package ... is already installed" in the middle of a line
+                    pkg_index = file_line.split().index("already") - 2
+                    pkg_name = file_line.split()[pkg_index].strip('"').rsplit("-",2)[0]
+                    required_pkgs.append(pkg_name)
+                    if pkg_index == 3:
+                        pkg_name = file_line.split()[7]
+                    else:
+                        pkg_name = file_line.split()[2]
+                    required_pkgs.append(pkg_name)
+
+                # TODO: len(file_line.split()) == 9 ??
+
+                elif len(file_line.split()) == 8 or len(file_line.split()) == 3:
+                    pkg_name = file_line.split()[2]
+                    required_pkgs.append(pkg_name)
+
+                elif len(file_line.split()) == 7 or len(file_line.split()) == 4:
+                    continue
+
+                elif len(file_line.split()) == 6 or len(file_line.split()) == 5:
+                    # DNF5 uses B/KiB/MiB/GiB, DNF4 uses B/k/M/G
+                    if file_line.split()[4] in ["B", "KiB", "k", "MiB", "M", "GiB", "G"]:
+                        continue
+                    else:
+                        pkg_name = file_line.split()[2]
+                        required_pkgs.append(pkg_name)
+
+                else:
+                    from content_resolver.exceptions import KojiRootLogError
+                    raise KojiRootLogError
+        
+
+        # 4/
+        # I'm done. So I can break out of the loop.
+        elif state == 4:
+            break
+            
+
+    return required_pkgs
+
+
+def _download_root_log_with_retry(root_log_url):
```

Comment:
This can be used for `Downloading the root.log file...` in `_resolve_srpm_using_root_log`.